cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java

```diff
@@ -454,4 +458,37 @@ protected static <R> R connectionResult(DSLContext dslContext, ConnectionCallabl
         }
     }
 
+    protected String parseLocFromTimeSeriesId(String tsId) {
+        String[] parts = tsId.split("\\.");
+        return parts[0];
+    }
+
+    @NotNull
+    protected VersionType getVersionType(String names, String office, boolean dateProvided) {
+        VersionType dateVersionType;
+
+        if (!dateProvided) {
+            boolean isVersioned = isVersioned(names, office);
+
+            if (isVersioned) {
+                dateVersionType = VersionType.MAX_AGGREGATE;
+            } else {
+                dateVersionType = VersionType.UNVERSIONED;
+            }
+
+        } else {
+            dateVersionType = VersionType.SINGLE_VERSION;
+        }
+
+        return dateVersionType;
+    }
+
+    private boolean isVersioned(String names, String office) {
+        return connectionResult(dsl, connection -> {
+            Configuration configuration = getDslContext(connection, office).configuration();
+            return OracleTypeMap.parseBool(CWMS_TS_PACKAGE.call_IS_TSID_VERSIONED(configuration,
+                    names, office));
+        });
+    }
+
```

Comment:
These all seem pretty version specific.  The JooqDao class doesn't feel like the right spot for them.

Reply:
yeah eithet TimseriesDao or a new Ts Utility class.

Reply:
Used by TimeseriesDao, TimeSeriesBinaryDao, and RegularTimeSeriesTextDao currently. Since there's no common base class there, utility class seems like a good candidate?

Reply:
Yeah. but I think also fine to just stick in TimesSeriesDAO. it's time series operation regardless of the underlying data storage.